test/jdk/java/net/httpclient/SpecialHeadersTest.java

```diff
@@ -453,10 +469,14 @@ void testAsync(String uriString, String headerNameAndValue, boolean sameClient)
         String value =  useDefault ? DEFAULTS.get(key).apply(uri) : v;
 
         HttpClient client = null;
+        Tracker tracker = null;
+        Throwable thrown = null;
         for (int i=0; i< ITERATION_COUNT; i++) {
             try {
-                if (!sameClient || client == null)
+                if (!sameClient || client == null) {
                     client = newHttpClient("testAsync", sameClient);
+                    tracker = TRACKER.getTracker(client);
```

Comment:
Should this be outside this `if` block? Otherwise, in its current form, I think this `tracker` can be `null` and then later in this test where we do `TRACKER.check(tracker, ...)`, I believe it will lead to an NPE inside the `check` implementation.

Reply:
Please disregard this comment. I took a more closer look and what you have here is fine, since we only call the `TRACKER.check(tracker,...)` when `sameClient = false` in which case this above `if` block would already have set the `tracker`.